src/seraphis_main/tx_validators.cpp

```diff
@@ -190,46 +190,28 @@ bool validate_sp_semantics_sp_reference_sets_v1(const SemanticConfigSpRefSetV1 &
     if (sp_membership_proofs.size() == 0)
         return true;
 
-    // check ref set decomp
-    const std::size_t ref_set_decomp_n{sp_membership_proofs[0].ref_set_decomp_n};
-    const std::size_t ref_set_decomp_m{sp_membership_proofs[0].ref_set_decomp_m};
+    // proof ref set decomposition (n^m) should match number of referenced enotes
+    const std::size_t ref_set_size{math::uint_pow(config.decomp_n, config.decomp_m)};
 
-    if (ref_set_decomp_n < config.decomp_n_min ||
-        ref_set_decomp_n > config.decomp_n_max)
-        return false;
-
-    if (ref_set_decomp_m < config.decomp_m_min ||
-        ref_set_decomp_m > config.decomp_m_max)
-        return false;
+    // bin width (2*radius+1)
+    const std::size_t bin_width{compute_bin_width(config.bin_radius)};
 
-    // check binned reference set configuration
-    const SpBinnedReferenceSetConfigV1 bin_config{sp_membership_proofs[0].binned_reference_set.bin_config};
-
-    if (bin_config.bin_radius < config.bin_radius_min ||
-        bin_config.bin_radius > config.bin_radius_max)
-        return false;
+    // num bin loci
+    const std::size_t num_bin_loci = sp_membership_proofs[0].bin_loci.size();
 
-    if (bin_config.num_bin_members < config.num_bin_members_min ||
-        bin_config.num_bin_members > config.num_bin_members_max)
+    // Check that the number of bin loci evenly divides the total reference set size
+    if (ref_set_size % num_bin_loci != 0)
```

Comment:
```suggestion
    if ((num_bin_loci == 0) || (ref_set_size % num_bin_loci != 0))
```
Potential division by zero here.